### PR TITLE
Run add capacity test via cli if there is a SSL ERROR

### PR DESCRIPTION
**Pull request by OdedViner**
Signed-off-by: Oded Viner <oviner@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 1]
```diff
@@ -60,3 +60,49 @@ def ui_deployment_conditions():
...
+    elif ocp_version not in ("4.7", "4.8"):
+        logger.info(
+            f"Add capacity via UI is not supported when the OCP version [{ocp_version}]"
+        )
+        return False
+    elif is_external or is_disconnected or is_proxy or is_lso:
+        if is_external:
+            logger.info("Add capacity via UI is not supported on external cluster")
+        if is_disconnected:
+            logger.info("Add capacity via UI is not supported on disconnected cluster")
```
Comment (hnallurv) [inline ocs_ci/ocs/ui/helpers_ui.py:101]: Shouldn't this message be "Add capacity via UI is not automated at the moment on disconnected cluster"?

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 2]
```diff
@@ -60,3 +60,49 @@ def ui_deployment_conditions():
...
+        if is_proxy:
+            logger.info("Add capacity via UI is not supported on proxy cluster")
```
Comment (hnallurv) [inline ocs_ci/ocs/ui/helpers_ui.py:103]: Shouldn't this message be "Add capacity via UI is not automated at the moment on proxy cluster"?

File: ocs_ci/ocs/ui/helpers_ui.py [hunk 3]
```diff
@@ -60,3 +60,49 @@ def ui_deployment_conditions():
...
+        if is_lso:
+            logger.info("Add capacity via UI is not supported on lso cluster")
```
Comment (hnallurv) [inline ocs_ci/ocs/ui/helpers_ui.py:105]: Shouldn't this message be "Add capacity via UI is not automated at the moment on LSO cluster"?

**Review by hnallurv — CHANGES_REQUESTED** [review]

File: ocs_ci/ocs/ui/helpers_ui.py (hunk 1 above)
Comment (OdedViner) [reply ocs_ci/ocs/ui/helpers_ui.py:101]: There is no difference between disconnected/external/proxy cluster and "regular" cluster (for UI automation), sometimes there is a SSL issue (at login UI step) on these type of clusters.

File: ocs_ci/ocs/ui/helpers_ui.py (hunk 2 above)
Comment (OdedViner) [reply ocs_ci/ocs/ui/helpers_ui.py:103]: There is no difference between disconnected/external/proxy cluster and "regular" cluster (for UI automation), sometimes there is a SSL issue (at login UI step) on these type of clusters.

File: ocs_ci/ocs/ui/helpers_ui.py (hunk 3 above)
Comment (OdedViner) [reply ocs_ci/ocs/ui/helpers_ui.py]: done

File: tests/manage/z_cluster/cluster_expansion/test_add_capacity.py [hunk 4]
```diff
@@ -38,7 +39,16 @@ def add_capacity_test():
...
     existing_osd_pod_names = [pod.name for pod in existing_osd_pods]
-    result = storage_cluster.add_capacity(osd_size)
+
+    if ui_add_capacity_conditions():
+        try:
+            result = add_capacity_ui(osd_size)
+        except Exception as e:
+            logging.error(e)
+            result = storage_cluster.add_capacity(osd_size)
+    else:
```
Comment (ebenahar) [inline tests/manage/z_cluster/cluster_expansion/test_add_capacity.py:49]: maybe worth adding a massage saying that add capacity from UI is not applicable and CLI method will be done

Comment (OdedViner) [reply tests/manage/z_cluster/cluster_expansion/test_add_capacity.py:50]: done

**Review by yitzhak12 — APPROVED** [review]
LGTM

**Review by ebenahar — APPROVED** [review]

File: ocs_ci/ocs/ui/helpers_ui.py (hunk 1 above)
Comment (hnallurv) [reply ocs_ci/ocs/ui/helpers_ui.py]: Resolved. Requested change has been incorporated.

File: ocs_ci/ocs/ui/helpers_ui.py (hunk 3 above)
Comment (hnallurv) [reply ocs_ci/ocs/ui/helpers_ui.py]: Resolved. Requested change has been incorporated.

**Review by hnallurv — APPROVED** [review]

**Pull request merged**